app/src/main/java/com/github/sdp_begreen/begreen/models/TrashCategory.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.sdp_begreen.begreen.models
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.github.sdp_begreen.begreen.R
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+
+enum class TrashCategory(val id : String, val title : String, val color : Float) : Parcelable {
+
+    // The title will be displayed in the screen to the user, so we prefer to define it in the strings.xml instead
+    // of Capitalizing the toString() value.
+
+    // The id is unique (stored in the database). We preferred to store a category by its "id" instead of its "value"
+    // name (i.e PLASTIC) because the value name could be changed afterward.
```

Comment:
> if we decide to change the title of an enum value of TrashCategory and we do not delete all the Bins that are associated with this enum value in the database, then we will fetch those bins with a new name that might be totally different to the real TrashCategory of the bins.

I agree with you on this point. The value we store on the database should be **immutable**. This is why I think that we should store its `id` and not its String representation.

What do you think about changing this : 

> We actually still store the String of the TrashCategory

to store its `id` ? As I said in previous comments (please see above), I don't think that storing a String representation is the right choice.

Reply:
I started to take another approach : remove the `id` field and identify an enum case with its default `name` attribute. Overriding the enum toString() method doesn't affect the name (i.e PLASTIC.name = "PLASTIC" no matter the toString()).

But finally I kept the `id` because I though about the following :

Each programming langage has its own conventions. In Kotlin (android) we use to declare enums in _THIS_FORMAT_ while in swift (iOS) its in _thisformat_. I think that since our app is developed in a native langage, we should take into consideration the fact that it can also be developed in Swift with its own conventions and rules.

Therefore, identifying a `TrashCategory` in the database with the enum case property applies constraints. This is why I recommend to keep a common `id` field. Let me know what you think about it